### ISSUE TITLE
I want to use cross-validation to train and predict all my own data, how to use eval.py?

### ISSUE THREAD

**Issue by jmfu95** [dennybritz/cnn-text-classification-tf#160]
My data has no test dataset. I want to use cross-validation to train and predict my data.Should I use my data to train, then use eval.py to predict?  If not, how can I get predicted label for all data? Someone knows? Thanks

**Comment by Soonmok** [comment]
you need to split your data into train_data and test_data 
you can use scikit-learn library to split it 
The function from library you need is train_test_split

Here is document
http://scikit-learn.org/stable/modules/generated/sklearn.model_selection.train_test_split.html

**Comment by Soonmok** [comment]
I just create pull request which uses cross-validation, and you can check it. (but its not for testing, its for training.)

**Comment by jmfu95** [comment]
Thanks for your reply. You mean I should use this function to split train and test then use eval.py, but I train the data in train.py. so if I use pre-trained model in this dataset, then I use this data to evaluate, it maybe get too high accuracy. Is that true?

**Comment by kaiwendt92** [comment]
Which pre-trained model you want to use ? A model that is not based on your own data ?

This is not a good idea, since you can't rely on the outcome.

Just take your data. Use the function from scikit learn to split your data into train and test data (80/20, 70/30 depending on how much you have) and then use this model in eval.py


________________________________
Von: jmfuuu <notifications@github.com>
Gesendet: Mittwoch, 25. Juli 2018 11:11
An: dennybritz/cnn-text-classification-tf
Cc: Subscribed
Betreff: Re: [dennybritz/cnn-text-classification-tf] I want to use cross-validation to train and predict all my own data, how to use eval.py? (#160)


Thanks for your reply. You mean I should use this function to split train and test then use eval.py, but I train the data in train.py. so if I use pre-trained model in this dataset, then I use this data to evaluate, it maybe get too high accuracy. Is that true?

**Comment by jmfu95** [comment]
OK. I understand. Thank you all.
